com.unity.ml-agents/Runtime/Inference/Utils/Multinomial.cs

```diff
@@ -32,10 +36,11 @@ public Multinomial(int seed)
         /// to be monotonic (always increasing). If the CMF is scaled, then the last entry in
         /// the array will be 1.0.
         /// </param>
-        /// <returns>A sampled index from the CMF ranging from 0 to cmf.Length-1.</returns>
-        public int Sample(float[] cmf)
+        /// <param name="numBranches">The number of possible branches, i.e. the effective size of the cmf array.</param>
+        /// <returns>A sampled index from the CMF ranging from 0 to numBranches-1.</returns>
+        public int Sample(float[] cmf, int numBranches)
```

Comment:
Because the float[] might be larger than we need now, we also pass the effective size of the array.

We could instead (or in additiona) repeat the final `sumProb` value in `ComputeCdf()`